src/Features/VisualBasic/Portable/CodeRefactorings/MoveType/VisualBasicMoveTypeService.vb

```diff
@@ -13,5 +16,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.CodeRefactorings.MoveType
         <ImportingConstructor>
         Public Sub New()
         End Sub
+
+        Protected Overrides Async Function GetRelevantNodeAsync(document As Document, textSpan As TextSpan, cancellationToken As CancellationToken) As Task(Of TypeBlockSyntax)
+            Dim someThing As TypeStatementSyntax = Await document.TryGetRelevantNodeAsync(Of TypeStatementSyntax)(textSpan, cancellationToken).ConfigureAwait(False)
+            Return TryCast(someThing?.Parent, TypeBlockSyntax)
```

Comment:
@CyrusNajmabadi @mavasani would like to get your thoughts on this

I noticed in the CI that VB tests were failing.  Some info you probably already know first.
VB has a different syntax tree structure for types, due to the fact that the class declaration has a block structure with a declaration ```Friend Class ...``` and an end ```End Class```

The tree for VB thus has the following structure.
```TypeBlockSyntax -> TypeStatementSyntax -> Identifier```
Whereas csharp has
```TypeDeclarationSyntax -> Identifier```

The original code is looking for the ```TypeBlockSyntax``` for VB.  The helper fails to find this when the caret is placed at the end of the class name.  This is because the parent of the parent of the identifier is the ```TypeBlockSyntax``` which ends somewhere down below the caret instead of at the caret, so [this check](http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/CodeRefactorings/AbstractRefactoringHelpersService.cs,209) fails.

So I fixed this by actually looking for the TypeStatementSyntax.  I did try just changing everything to use TypeStatementSyntax, but the abstract move type implementation has other bits and pieces that depend on the type declaration not being a child of the actual type declaration block (e.g. IsNestedType, TopLevelTypeDeclarations plus a bunch more failing tests even after fixing that).

I also thought about changing the helper, but this seems more like an issue of searching for the right node than an actual problem in the helper.  And I didn't see an easy way to fix this without modifying the behavior negatively for all block syntax nodes.

Reply:
> I also thought about changing the helper, but this seems more like an issue of searching for the right node than an actual problem in the helper. And I didn't see an easy way to fix this without modifying the behavior negatively for all block syntax nodes.

This is a good question.  If @petrroll is still interactin with Roslyn, i'd like to hear his thoughts.

I think special knowledge of VB (and this interesting statement/block relation) might be fine for the helper method.  

However, i'm also 100% ok with the approach you took here to just isolate this to the feature.  My intuition is that the helper should adopt this approach.  however, i'm not an expert there and would defer to Petr.

For example, the IRefactoringHelpersService already has several special cases for many common language patterns (like the relation between expressions and expression-statements).  This morally feels around the same to me.  But, until we get resolution on this topic, i def think your PR approach here is 100% ok!